torch/_functorch/_aot_autograd/runtime_wrappers.py

```diff
@@ -1865,8 +1872,19 @@ def get_types_for_tangents(tangents):
                         )
                         for i, t in enumerate(all_args)
                     ]
+
+                    metas = iter(CompiledFunction.metadata.subclass_tangent_meta)
+                    tangent_metadata = [
+                        next(metas)
+                        if isinstance(a, Tensor) and is_traceable_wrapper_subclass(a)
+                        else i
+                        for i, a in enumerate(all_args)
+                    ]
+
                     all_args = unwrap_tensor_subclasses(
-                        all_args, is_joint_structure=False
+                        all_args,
+                        is_runtime=False,
```

Comment:
I'm a bit confused - we're setting `is_runtime=False` here, but this is definitely (backward) runtime. Is this incorrect? Or is it doing the right thing, but this flag is named incorrectly?

Reply:
tbh that seems like a good reason to split it out and have a separate function for the runtime path, instead of keeping the `is_runtime` flag around (that will also make optimizing the runtime code easier later on)

Reply:
Split the code between `unwrap_tensor_subclass` and `runtime_unwrap_tensor_subclass`